internal/bminventory/inventory.go

```diff
@@ -882,6 +886,32 @@ func (b *bareMetalInventory) createAndUploadDay2NodeIgnition(ctx context.Context
 	return nil
 }
 
+func (b *bareMetalInventory) UpdateIgnitionEndpointIfHasMCSCert(log logrus.FieldLogger, host *models.Host, cluster *common.Cluster, ignitionEndpointUrl string) (string, error) {
+	// No ignition override set by bmh agent controller
+	if host.IgnitionConfigOverrides == "" {
+		return ignitionEndpointUrl, nil
+	}
+	// No CA certs set in ignition override
+	if !ignition.HasCACertInIgnition(host.IgnitionConfigOverrides) {
```

Comment:
Is there any chance that we're providing a CACert for some purpose other than the spoke cluster?
Worried that this could cause failures if someone needs a cert for some other reason.

Reply:
It has to be a host-specific CA override (cluster-wide Ignition with proxy CA won't trigger this). We know of only one usecase for this - a custom Ignition service and this PR won't break it